diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index fd0669f..32fdc8e 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -16,13 +16,6 @@ else ()
   set (MSVC_TOOLSET_VERSION 0)
   set (MSVC_TOOLSET_MAJOR 0)
 endif ()
-if (CMAKE_CROSSCOMPILING)
-  # Ensure that all "true" (resp. "false") settings are represented by
-  # the same string.
-  set (CMAKE_CROSSCOMPILING_STR "ON")
-else ()
-  set (CMAKE_CROSSCOMPILING_STR "OFF")
-endif ()
 
 # geographiclib-config.cmake for the build tree
 set (PROJECT_ROOT_DIR "${PROJECT_BINARY_DIR}")
diff --git a/cmake/project-config-version.cmake.in b/cmake/project-config-version.cmake.in
index f15702f..2c731b7 100644
--- a/cmake/project-config-version.cmake.in
+++ b/cmake/project-config-version.cmake.in
@@ -14,13 +14,6 @@ if (MSVC)
   # digit.
   math (EXPR MSVC_TOOLSET_MAJOR "${MSVC_TOOLSET_VERSION}/10")
 endif ()
-if (CMAKE_CROSSCOMPILING)
-  # Ensure that all "true" (resp. "false") settings are represented by
-  # the same string.
-  set (CMAKE_CROSSCOMPILING_STR "ON")
-else ()
-  set (CMAKE_CROSSCOMPILING_STR "OFF")
-endif ()
 
 if (NOT PACKAGE_FIND_NAME STREQUAL "@PROJECT_NAME@")
   # Check package name (in particular, because of the way cmake finds
@@ -43,10 +36,6 @@ elseif (MSVC AND NOT (
   # Reject if there's a mismatch in MSVC compiler versions
   set (REASON "MSVC_TOOLSET_VERSION = @MSVC_TOOLSET_VERSION@")
   set (PACKAGE_VERSION_UNSUITABLE TRUE)
-elseif (NOT CMAKE_CROSSCOMPILING_STR STREQUAL "@CMAKE_CROSSCOMPILING_STR@")
-  # Reject if there's a mismatch in ${CMAKE_CROSSCOMPILING}
-  set (REASON "cross-compiling = @CMAKE_CROSSCOMPILING@")
-  set (PACKAGE_VERSION_UNSUITABLE TRUE)
 elseif (CMAKE_CROSSCOMPILING AND
     NOT (CMAKE_SYSTEM_NAME STREQUAL "@CMAKE_SYSTEM_NAME@" AND
       CMAKE_SYSTEM_PROCESSOR STREQUAL "@CMAKE_SYSTEM_PROCESSOR@"))
